logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueWriterTest.java

```diff
@@ -299,8 +299,6 @@ public void testRemoveOldestSegmentWhenRetainedSizeIsExceededAndDropOlderModeIsE
     @Test
     public void testRemoveSegmentsOrder() throws IOException {
         try (DeadLetterQueueWriter sut = new DeadLetterQueueWriter(dir, 10 * MB, 20 * MB, Duration.ofSeconds(1))) {
-            Files.delete(dir.resolve("1.log.tmp"));
```

Comment:
Thanks for quick fix @andsel!
Shouldn't we still remove the temp since we don't use it? My concern is when we create a file and if it is `append` mode by default, it may bring disk/inode usage issue in the future. But since my knowledge limited for this case, plz give me details. Thanks!

Reply:
> Shouldn't we still remove the temp since we don't use it?

Indeed it's actively used, it's the writer side of the DLQ. When the segment file is full, or Logstash process is restarted, then it's moved from `1.log.tmp` to `1.log` and is considered available to be consumed by the reader.

Reply:
Yep, moved into PR #14073